src/sheet_metal_conus.py

```diff
@@ -77,6 +96,71 @@ def getColorString(self, longColor, verbose=False):
         return hexColor
 
+    # Marker arrows
+    def makeMarkerstyle(self, name, rotate):
+        defs = self.xpathSingle('/svg:svg//svg:defs')
+        if defs == None:
+            defs = inkex.etree.SubElement(self.document.getroot(),inkex.addNS('defs','svg'))
+        marker = inkex.etree.SubElement(defs ,inkex.addNS('marker','svg'))
+        marker.set('id', name)
+        marker.set('orient', 'auto')
+        marker.set('refX', '0.0')
+        marker.set('refY', '0.0')
+        marker.set('style', 'overflow:visible')
+        marker.set(inkex.addNS('stockid','inkscape'), name)
+
+        arrow = inkex.etree.Element("path")
+        # messy but works; definition of arrows in beautiful DIN-shapes:
+        if name.startswith('ArrowDIN-'):
+            if rotate:
+                arrow.set('d', 'M 8,0 -8,2.11 -8,-2.11 z')
+            else:
+                arrow.set('d', 'M -8,0 8,-2.11 8,2.11 z')
+        if name.startswith('ArrowDINout-'):
+            if rotate:
+                arrow.set('d', 'M 0,0 16,2.11 16,0.5 26,0.5 26,-0.5 16,-0.5 16,-2.11 z')
+            else:
+                arrow.set('d', 'M 0,0 -16,2.11 -16,0.5 -26,0.5 -26,-0.5 -16,-0.5 -16,-2.11 z')
+        arrow.set('style', 'fill:#000000;stroke:none')
+        marker.append(arrow)
+
+    def set_arrow_dir(self, option):
+        if option=='inside':
+            # inside
+            self.arrowlen = 6.0
+            self.dimline_style['marker-start'] = 'url(#ArrowDIN-start)'
+            self.dimline_style['marker-end'] = 'url(#ArrowDIN-end)'
+            self.makeMarkerstyle('ArrowDIN-start', False)
+            self.makeMarkerstyle('ArrowDIN-end', True)
+        else:
+            # outside
+            self.arrowlen = 0
+            self.dimline_style['marker-start'] = 'url(#ArrowDINout-start)'
+            self.dimline_style['marker-end'] = 'url(#ArrowDINout-end)'
+            self.makeMarkerstyle('ArrowDINout-start', False)
+            self.makeMarkerstyle('ArrowDINout-end', True)
+
+    
+    def drawDimension(self, a, b, style, parent):
+        " draw arrowed dimensions using markers "
+        # draw arrows as inside or outside dimension
+        critical_length = 35.
+        if calc_dist_between_points(a,b) > critical_length:
+            self.set_arrow_dir('inside')
+        else:
+            self.set_arrow_dir('outside')
+        attribs = {'style' : simplestyle.formatStyle(style)}
+        # account for length change so arrows fit
+        norm = normalize(a, b)
+        dim_start_x = a[0] + self.arrowlen*norm[0]
+        dim_start_y = a[1] + self.arrowlen*norm[1]
+        dim_end_x = b[0] - self.arrowlen*norm[0]
+        dim_end_y = b[1] - self.arrowlen*norm[1]
+        #
+        attribs['d'] = 'M %f,%f %f,%f' % (dim_start_x, dim_start_y, dim_end_x, dim_end_y)
+        dimline = inkex.etree.SubElement(parent, inkex.addNS('path', 'svg'), attribs)
+        return dimline
+        
```

Comment:
Draw the dimension as a line with arrow at either end. color in style as well as arrow markers. Here we shorten the line to account for the length of the arrow markers....